spring-cloud-stream-core-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc

```diff
@@ -560,10 +520,44 @@ public static class TestPojoWithAnnotatedArguments {
 }
 ----
 
-NOTE: Dispatching through `@StreamListener` conditions is only supported for handlers of individual messages, not for reactive programming support (described spring-cloud-stream-overview-reactive-programming-support[later]).
+*Content Type Negotiation in the context of `condition`*
+
+It is important to understand some of the mechanics behind content-based routing using `condition` argument of `@StreamListener`, especially in the context of type of the message as a whole.
+It may also help if you familiarise yourself with the <<Content Type Negotiation>>  before you proceed.
+
+Consider the following scenario:
+
+[source,java]
+----
+@EnableBinding(Sink.class)
+@EnableAutoConfiguration
+public static class CatsAndDogs {
+
+    @StreamListener(target = Sink.INPUT, condition = "payload.class.simpleName=='Dog'")
+    public void bark(Dog dog) {
+       // handle the message
+    }
+
+    @StreamListener(target = Sink.INPUT, condition = "payload.class.simpleName=='Cat'")
+    public void purr(Cat cat) {
+       // handle the message
+    }
+}
+----
+
+The preceeding code is perfectly valid. It will compile and deploy without any issues, yet it will never produce the result you're expecting.
+That is because you are testing something that does not yet exist in a state you expect and is becouse the payload of the Message is not yet converted from the wire format (`byte[]`)to the desired type.
+In other words it has not yet gone through the type conversion process described in the <<Content Type Negotiation>>.
+
+So, unless you are using SPeL expression that evaluates raw data (for example, the value of the first byte in the byte array), use message header
+based expressions `condition = "headers['type']=='dog'"`
```

Comment:
Remove this line. I blended it with the preceding line to get the hyphen in the right place.